std/range/package.d

```diff
@@ -8647,3 +8647,237 @@ if (is(typeof(fun) == void) || isSomeFunction!fun)
 
     auto r = [1, 2, 3, 4].tee!func1.tee!func2;
 }
+
+template CommonElementType(Rs...)
+{
+    alias CommonElementType = CommonType!(staticMap!(ElementType, Rs));
+}
+
+alias isSortedRange(R) = isInstanceOf!(SortedRange, R); // TODO Or use: __traits(isSame, TemplateOf!R, SortedRange)
+
+/**
+   Merge several sorted ranges $(D rs) with less-than predicate function $(D
+   pred) into one single sorted range containing the sorted union of the
+   elements of inputs.
+
```

Comment:
Are you sure you want to allow mutable access to the elements of the original ranges via `front` and `back`?

Because if you mutate an element in the original range it may not be sorted anymore.